### DISCUSSION TITLE
Question about Ecole

### DISCUSSION THREAD

**Discussion by yanglinxin0612** [ds4dm/ml4co-competition#75] [Q&A]
Hi folks,

I just noticed that, for some instances, the dual bound obtained by Ecole is much lower than the one given in the .json file. Is this as intended, or am I obtaining the dual bound incorrectly? I was using model.dual_bound in my observation function to extract the current dual bound. @gasse Lord Bender please send help.

Thanks in advance
Best,
Linx

**Comment by gasse** [comment]
Hi @yanglinxin0612 ,

I think you're referring to the fact that the precomputed bounds (primal and dual) which we provide alongside each instance, do not necessarily match the initial bounds found by Ecole, after `env.reset()` for example. This is expected to happen. We precomputed those bounds with a certain random seed, which affect different parts of the solving process (such as preprocessing), and in turn affects the initial bounds found by SCIP (first feasible solution for the primal bound, and root LP solution value for the dual bound). In the way we run the evaluations and compute the rewards this mismatch should be handled properly. We basically take the min (primal) or max (dual) of the pre-computed initial bounds and the current bound to compute the integrals. Then since all participants are evaluated using the same random seeds,  all participants will actually use the same initial primal and dual bounds.

Hope that clarifies things, let us know if you have more questions.

Best,
Maxime
